src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs

```diff
@@ -36,24 +37,52 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 var contentItem = token.ToObject<ContentItem>();
                 var modifiedUtc = contentItem.ModifiedUtc;
                 var publishedUtc = contentItem.PublishedUtc;
+                var isPublished = contentItem.Published;
+                var isLatest = contentItem.Latest;
+
                 var existing = await _contentManager.GetVersionAsync(contentItem.ContentItemVersionId);
-                
+
                 if (existing == null)
                 {
                     // Initializes the Id as it could be interpreted as an updated object when added back to YesSql
                     contentItem.Id = 0;
                     await _contentManager.CreateAsync(contentItem);
-                    
+
                     // Overwrite ModifiedUtc & PublishedUtc values that handlers have changes
                     // Should not be necessary if IContentManager had an Import method
                     contentItem.ModifiedUtc = modifiedUtc;
                     contentItem.PublishedUtc = publishedUtc;
-                }
-                else
-                {
-                    // Replaces the id to force the current item to be updated
-                    existing.Id = contentItem.Id;
-                    _session.Save(existing);
+
+                    // Reset published and latest to imported values as CreateAsync sets these values arbitrarily.
+                    contentItem.Published = isPublished;
+                    contentItem.Latest = isLatest;
+
+                    // Resolve previous published or draft items or they will continue to be listed as published or draft.
+                    var relatedContentItems = await _session
+                        .Query<ContentItem, ContentItemIndex>(x =>
+                            x.ContentItemId == contentItem.ContentItemId && (x.Published || x.Latest))
+                        .ListAsync();
+
+                    // Alter previous items depending on published and latest values of imported item.
+                    foreach (var relatedItem in relatedContentItems)
+                    {
+                        // CreateAsync calls session.Save so the importing item is now resolved as part of the query.
+                        if (String.Equals(relatedItem.ContentItemVersionId, contentItem.ContentItemVersionId)){
+                            continue;
+                        }
+
+                        if (isPublished)
```

Comment:
Why not use contentItem.Published directly?

Reply:
Because depending on what is done in the `CreateAsync` or handlers, the value can be mutated. 
But here we should to force it to what is supplied in the recipe, so that the result is always the same (and is a defined, understandable result, that is idempotent, because we define what should happen going in, and always get the same output)

Reply:
Ok I agree too